src/gui/components/record/conflict/conflictfield.tsx

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2021, 2022 Macquarie University
+ *
+ * Licensed under the Apache License Version 2.0 (the, "License");
+ * you may not use, this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND either express or implied.
+ * See, the License, for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Filename: RecordTabBar.tsx
+ * Description:
+ *   File contains element of conflict field
+ *    ---- Conflict Field: Form field, Annoation with uncertiy , Updated By
+ *    ---- Empty Field before data Sync to device
+ *    ---- ConflictResolveIcon: shows number of conflict, unresolved , resolved and rejects
+ */
+
+import React from 'react';
+
+import {
+  Grid,
+  Box,
+  Card,
+  CardHeader,
+  CardContent,
+  IconButton,
+  Typography,
+} from '@mui/material';
+import ErrorOutlineOutlinedIcon from '@mui/icons-material/ErrorOutlineOutlined';
+import {AnnotationField} from '../Annotation';
+import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';
+import InfoIcon from '@mui/icons-material/Info';
+import CheckBoxIcon from '@mui/icons-material/CheckBox';
+import {cardsstyles, cardstyletype} from './conflictstyle';
+import {CircularProgress} from '@mui/material';
+import {getComponentFromFieldConfig} from '../fields';
+type FieldWithAnnotationProp = {
+  fieldName: string;
+  fieldConfig: any;
+  formProps: any;
+  styletype: string;
+  type?: string;
+  data?: any;
+};
+
+type EmptyProps = {
+  isspin: boolean;
+  text?: string;
+  isloading?: boolean;
+};
+export function EmptyField(props: EmptyProps) {
+  return (
+    <Box pt={10} px={3} style={cardsstyles.empty.card}>
+      {props.isspin && (
+        <Grid
+          container
+          justifyContent="center"
+          alignItems="center"
+          style={cardsstyles.empty.cardheader}
+        >
+          <Grid
+            item
+            xs={2}
+            container
+            justifyContent="center"
+            alignItems="flex-start"
+          >
+            {props.isloading === false ? (
+              <CircularProgress size={12} thickness={4} />
+            ) : (
+              <InfoOutlinedIcon style={cardsstyles.empty.iconstyle} />
+            )}
+          </Grid>
+
+          <Grid item xs={10}>
+            <Typography variant="caption" display="block">
+              {props.isloading === false
+                ? " This data isn't on your device yet. It's loading,it might take some time. You can wait or select another conflict to resolve in the meantime. "
```

Comment:
"This data isn't on your device yet. It's loading, but can take some time - depending on the data in conflict. You can wait, or select another conflict to resolve in the meantime (the conflict will continue to download in the background)."

This should be a conditional comment. If there are no more alternative conflicts to resolve, instead the comments should be:
"This data isn't on your device yet. It's loading, but can take some time - depending on the data in conflict."

Reply:
Thank you @lmannering for catching this. Somehow my tired brain thought these changes had been already committed this morning. I appreciate your attention to detail. Much obliged.